exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -110,11 +111,12 @@
   private LinkedList<BatchGroup> batchGroups = Lists.newLinkedList();
   private LinkedList<BatchGroup> spilledBatchGroups = Lists.newLinkedList();
   private SelectionVector4 sv4;
-  private FileSystem fs;
+  private static FileSystem fs;
```

Comment:
It was my first decision to put `closeSpillFileSystem` into close(). But it in this case we can have failed query when one leaf fragment executes `close()` and delete spill folder while another leaf fragment still spooling to the disk.
That's why we need to delete the spill folder only when root fragment (foreman) is closed.
And `fs` is initialized only once that's why every instance of class uses the same FileSystem.
If I am right one drillbit can perform only one query at the same time. And another drillbit performing an external sort uses own static FileSystem instance.